### PR TITLE
Create route groups #93

### PR DESCRIPTION
**Pull request by mapeveri**

### CODE REVIEW

**Comment by mapeveri** [comment]
Ready :)

File: masonite/helpers/routes.py [hunk 1]
```diff
@@ -34,3 +34,16 @@ def patch(url, controller):
     Shortcut for PATCH http
     """
     return Patch().route(url, controller)
+
+
+def group(url, route_list):
+    """
+    Group route
+    """
+    if url[-1:] != '/':
```
Comment (josephmancuso) [inline masonite/helpers/routes.py]: what is this here? this looks really confusing. Are you checking if the url starts with a forward slash?

If so you should just do something like:

```python
if not url.startswith('/'):
    # do logic
```

but even still I don't think we should check for this. We don't check in any other routes. All routes should start with a slash anyway. If we start doing this check here then we need to perform this check on all routes. If we are going to make this change then I say we do it in some 2.x release instead.

File: routes/web.py [hunk 2]
```diff
@@ -1,7 +1,12 @@
+from masonite.helpers.routes import group
 from masonite.routes import Get
 
 
 ROUTES = [
     Get().route('/test', None).middleware('auth'),
     Get().domain('test').route('/test', None).middleware('auth'),
+    group('/example', [
+        Get().route('test/1', 'TestController@show'),
```
Comment (josephmancuso) [inline routes/web.py]: this route should here should have forward slashes in the beginning. let's keep it in line with the rest of the routes

File: tests/test_routes.py [hunk 3]
```diff
@@ -76,5 +77,16 @@ def test_route_gets_controllers():
...
     assert Get().route('test/url', '/app.http.test_controllers.TestController@show')
 
+
 def test_route_doesnt_break_on_incorrect_controller():
     assert Get().route('test/url', 'BreakController@show')
+
+
+def test_group_route():
+    routes = group('/example', [
+        Get().route('test/1', 'TestController@show'),
```
Comment (josephmancuso) [inline tests/test_routes.py]: same thing here. Change this line to `Get().route('/test/1', 'TestController@show')` and get this same test to pass. Good test.

File: masonite/helpers/routes.py (hunk 1 above)
Comment (mapeveri) [reply masonite/helpers/routes.py]: No, I only check if it has "/" at the end. I erase it?

File: routes/web.py (hunk 2 above)
Comment (mapeveri) [reply routes/web.py]: Ok

File: tests/test_routes.py (hunk 3 above)
Comment (mapeveri) [reply tests/test_routes.py]: Ok

File: masonite/helpers/routes.py (hunk 1 above)
Comment (josephmancuso) [reply masonite/helpers/routes.py]: oh ok ummm then that's fine actually. But then do `url.endswith('/')`

Comment (josephmancuso) [reply masonite/helpers/routes.py]: actually you are appending a `/` here. Don't do that. Just get rid of this actually

Comment (mapeveri) [reply masonite/helpers/routes.py]: Ok

**Comment by mapeveri** [comment]
Check please :)

**Comment by josephmancuso** [comment]
Looks awesome! I'm gonna make some minor variable name changes and merge this in

**Pull request merged**